cairo/ethereum/cancun/vm/instructions.cairo

```diff
@@ -0,0 +1,703 @@
+    signed_greater_than,
+    equal,
+    is_zero,
+)
+from ethereum.cancun.vm.instructions.bitwise import (
+    bitwise_and,
+    bitwise_or,
+    bitwise_xor,
+    bitwise_not,
+    get_byte,
+    bitwise_shl,
+    bitwise_shr,
+    bitwise_sar,
+)
+from ethereum.cancun.vm.instructions.keccak import keccak
+from ethereum.cancun.vm.instructions.block import (
+    block_hash,
+    coinbase,
+    timestamp,
+    number,
+    prev_randao,
+    gas_limit,
+    chain_id,
+)
+from ethereum.cancun.vm.instructions.control_flow import stop, jump, jumpi, pc, gas_left, jumpdest
+from ethereum.cancun.vm.instructions.storage import sload, sstore, tload, tstore
+from ethereum.cancun.vm.instructions.stack_instructions import pop
+from ethereum.cancun.vm.instructions.stack_instructions import (
+    push0,
+    push1,
+    push2,
+    push3,
+    push4,
+    push5,
+    push6,
+    push7,
+    push8,
+    push9,
+    push10,
+    push11,
+    push12,
+    push13,
+    push14,
+    push15,
+    push16,
+    push17,
+    push18,
+    push19,
+    push20,
+    push21,
+    push22,
+    push23,
+    push24,
+    push25,
+    push26,
+    push27,
+    push28,
+    push29,
+    push30,
+    push31,
+    push32,
+)
+from ethereum.cancun.vm.instructions.stack_instructions import (
+    dup1,
+    dup2,
+    dup3,
+    dup4,
+    dup5,
+    dup6,
+    dup7,
+    dup8,
+    dup9,
+    dup10,
+    dup11,
+    dup12,
+    dup13,
+    dup14,
+    dup15,
+    dup16,
+)
+from ethereum.cancun.vm.instructions.stack_instructions import (
+    swap1,
+    swap2,
+    swap3,
+    swap4,
+    swap5,
+    swap6,
+    swap7,
+    swap8,
+    swap9,
+    swap10,
+    swap11,
+    swap12,
+    swap13,
+    swap14,
+    swap15,
+    swap16,
+)
+from ethereum.cancun.vm.instructions.memory_instructions import mstore, mstore8, mload, msize, mcopy
+from ethereum.cancun.vm.instructions.log import log0, log1, log2, log3, log4
+from ethereum.cancun.vm.instructions.system import revert
+from ethereum.cancun.vm.instructions.environment import (
+    address,
+    balance,
+    origin,
+    caller,
+    callvalue,
+    calldataload,
+    calldatasize,
+    calldatacopy,
+    codesize,
+    codecopy,
+    gasprice,
+    extcodesize,
+    extcodecopy,
+    returndatasize,
+    returndatacopy,
+    extcodehash,
+    self_balance,
+    base_fee,
+    blob_hash,
+    blob_base_fee,
+)
+
+func op_implementation{
+    range_check_ptr,
+    bitwise_ptr: BitwiseBuiltin*,
+    keccak_ptr: KeccakBuiltin*,
+    poseidon_ptr: PoseidonBuiltin*,
+    evm: Evm,
+}(opcode: felt) -> EthereumException* {
+    // call opcode
+    // count 1 for "next line" and 4 steps per opcode: call, opcode, jmp, end
+    tempvar offset = opcode * 4 + 1;
+
+    // Prepare arguments
+    [ap] = range_check_ptr, ap++;
+    [ap] = bitwise_ptr, ap++;
+    [ap] = keccak_ptr, ap++;
+    [ap] = poseidon_ptr, ap++;
+    [ap] = evm.value, ap++;
+
+    jmp rel offset;
+    call stop;  // 0x0
+    jmp end;
+    call add;  // 0x1 - ADD
```

Comment:
could just `ret` after each opcode instead of rebinding at the end?

Reply:
done